src/plugins/intel_gpu/include/intel_gpu/primitives/border.hpp

```diff
@@ -38,39 +38,57 @@ struct border : public primitive_base<border> {
     /// @param output_padding     Optional padding for output from primitive.
     border(const primitive_id& id,
            const input_info& input,
-           const ov::CoordinateDiff& pads_begin = {0, 0, 0, 0},
-           const ov::CoordinateDiff& pads_end = {0, 0, 0, 0},
+           const std::vector<int64_t>& pads_begin = {},
+           const std::vector<int64_t>& pads_end = {},
            const ov::op::PadMode pad_mode = ov::op::PadMode::CONSTANT,
            const float pad_value = 0.0f,
            const padding& output_padding = padding())
         : primitive_base(id, {input}, {output_padding}),
           pads_begin(pads_begin),
           pads_end(pads_end),
           pad_mode(pad_mode),
-          pad_value(pad_value) {}
+          pad_value(pad_value),
+          pad_value_input_constant(true) {}
 
     /// @brief Constructs border primitive / layer with dynamic pads.
     border(const primitive_id& id,
-           const input_info& input,
-           const input_info& pads_begin_id,
-           const input_info& pads_end_id,
+           const std::vector<input_info>& inputs,
+           const std::vector<int64_t>& pads_begin = {},
+           const std::vector<int64_t>& pads_end = {},
            const ov::op::PadMode pad_mode = ov::op::PadMode::CONSTANT,
            const float pad_value = 0.0f,
            const padding& output_padding = padding())
-        : primitive_base(id, {input, pads_begin_id, pads_end_id}, {output_padding}),
-          pads_begin({}),
-          pads_end({}),
+        : primitive_base(id, inputs, {output_padding}),
+          pads_begin(pads_begin),
+          pads_end(pads_end),
           pad_mode(pad_mode),
-          pad_value(pad_value) {}
+          pad_value(pad_value),
+          pad_value_input_constant(true) {}
+
+    /// @brief Constructs pad_mode is PadMode::CONSTANT and pad_value comes from parameter input.
+    border(const primitive_id& id,
+           const std::vector<input_info>& inputs,
+           const std::vector<int64_t>& pads_begin,
+           const std::vector<int64_t>& pads_end,
+           const padding& output_padding)
+        : primitive_base(id, inputs, {output_padding}),
+          pads_begin(pads_begin),
+          pads_end(pads_end),
+          pad_mode(ov::op::PadMode::CONSTANT),
+          pad_value(0.0f),
+          pad_value_input_constant(false) {}
 
     /// @brief Sizes of border that needs to be added from left (in X dimension) and from top (in Y dimension).
-    ov::CoordinateDiff pads_begin;
+    std::vector<int64_t> pads_begin;
```

Comment:
Why did you change this to vector? 
This was adjusted to ov type ..

Reply:
done